x/emissions/module/rewards/rewards.go

```diff
@@ -427,14 +428,32 @@ func payoutRewards(ctx sdk.Context, k keeper.Keeper, rewards []TaskRewards) erro
 			return errors.Wrapf(err, "failed to decode payout address")
 		}
 
-		err = k.BankKeeper().SendCoinsFromModuleToAccount(
-			ctx,
-			types.AlloraRewardsAccountName,
-			address,
-			sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, reward.Reward.SdkIntTrim())),
-		)
-		if err != nil {
-			return errors.Wrapf(err, "failed to send coins from rewards module to payout address")
+		if reward.Reward.IsZero() {
+			continue
+		}
+
+		if reward.Type == ReputerRewardType {
+			oneE18, err := inference_synthesis.AlloraOneE18()
+			if err != nil {
+				return err
+			}
+			scaledReward, err := reward.Reward.Mul(oneE18)
```

Comment:
Why do we multiply by `oneE18` here and not below?

Why don't we divide by `oneE18` at an earlier step, from which `reward` is derived?

Reply:
Because rewards are calculated as a decimal number in the rest of the system I believe. So if someones rewards are 1.2 allo, then reward.Reward is 1.2 allo, so we wouldn't need to divide by 1e18. However we do need to muliply by 1e18 (and the Math.floor the result) to get the number of tokens to send out or add to stake

Reply:
and yes, should have multiplied in both places. I updated.

Reply:
at no point in any of the reward code I wrote nor any of the code I think guilherme wrote do we ever operate with allo


all computations are done with the base unit of uAllo

I don't know why we need to do any multiplication or division by 1e18, ever. I don't think it's necessary in any point in the codebase

Reply:
I think so too

Reply:
removed